addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/internal/core/LxControlColorPickerV2.java

```diff
@@ -0,0 +1,294 @@
+     */
+    public void on() throws IOException {
+        HSBType currentColor = this.getColor();
+        if (currentColor != null) {
+            this.setColor(new HSBType(currentColor.getHue(), currentColor.getSaturation(), PercentType.HUNDRED));
+        }
+    }
+
+    /**
+     * Sets the color picker to off
+     *
+     * @throws IOException
+     *                         error communicating with the Miniserver
+     */
+    public void off() throws IOException {
+        HSBType currentColor = this.getColor();
+        if (currentColor != null) {
+            this.setColor(new HSBType(currentColor.getHue(), currentColor.getSaturation(), PercentType.ZERO));
+        }
+    }
+
+    /**
+     * set the brightness level
+     *
+     * @param p the brightness percentage
+     */
+    public void setBrightness(PercentType p) {
+        HSBType currentColor = this.getColor();
+        if (currentColor != null) {
+            this.setColor(new HSBType(currentColor.getHue(), currentColor.getSaturation(), p));
+        }
+    }
+
+    /**
+     * set the brightness level from a decimal type
+     *
+     * @param d the brightness in decimal
+     */
+    public void setBrightness(DecimalType d) {
+        this.setBrightness(new PercentType(d.toBigDecimal()));
+    }
+
+    /**
+     * increases/decreases the brightness with a given step
+     *
+     * @param step the amount to increase/decrease
+     */
+    public void increaseDecreaseBrightness(int step) {
+        HSBType currentColor = this.getColor();
+        if (currentColor != null) {
+            this.setBrightness(new PercentType(currentColor.getBrightness().intValue() + step));
+        }
+    }
+
+    /**
+     * map the Loxone color to OpenHab HSBType
+     *
+     * @param color color in format hsb(h,s,v) or temp(brightness,kelvin)
+     * @return HSBType
+     */
+    private HSBType mapLoxoneToOH(String color) {
+        HSBType hsbColor = null;
+
+        Matcher m = Pattern.compile("\\((.*?)\\)").matcher(color);
+        if (m.find()) {
+            String[] colorSplit = m.group(1).split(",");
+
+            if (isTemperatureColor(color) && colorSplit.length == 2) {
+                // convert temperature to a OH color
+                int brightness = constrain(Integer.valueOf(colorSplit[0]), 0, 100);
+                int temperature = constrain(Integer.valueOf(colorSplit[1]), 0, 65500);
+
+                int red = map(brightness, 0, 100, 0, calculateRed(temperature));
+                int green = map(brightness, 0, 100, 0, calculateGreen(temperature));
+                int blue = map(brightness, 0, 100, 0, calculateBlue(temperature));
+
+                hsbColor = HSBType.fromRGB(red, green, blue);
+            } else if (isHsvColor(color) && colorSplit.length == 3) {
+                hsbColor = new HSBType(m.group(1));
+            } else {
+                throw new IllegalArgumentException("Invalid color: " + color);
```

Comment:
This exception is not caught in the caller of this method. I suggest to return null and to catch this exception from new HSBType in this method and return null too.